Added configurable automatic fixing and docker support

indy2kro opened the pull request:
Added multiple enhancements for the running and automatic fixing for hooks:

- automatically_fix_errors: automatically run the fixer without any CLI prompt
- rerun_analyzer_after_autofix: automatically re-run the analyzer command after autofix (by default it only runs once before the fix)
- stop_at_first_analyzer_failure: stop (or not) at the first analyzer failure encountered - useful when many hooks are configured to see (and maybe fix) all the issues instead of running multiple times

Added various debug features:

- output_errors: output any errors encountered during execution directly for easy debug
- debug_commands: display the commands that are executed
- debug_output: display the output of each command during execution directly for easy debug

Added support for docker execution for hooks - each hook can be configured independently using `run_in_docker` and `docker_container` parameter.

Small enhancement for file_extensions parameter on Larastan hook (was probably missed, it should only run for php files).

The default values for the new parameters are set so that they are compatible with the previous version behavior, for optimal usage would probably want to set to something like:

 - output_errors: true
 - automatically_fix_errors: true
 - rerun_analyzer_after_autofix: true
 - stop_at_first_analyzer_failure: false

Added unit tests for some of the functionality, unfortunately I was not able to get 100% code coverage though.

Also added dev deps for `pestphp/pest-plugin-laravel` for some better auto-completion and extra functionalities (not used for the new tests though).

Commit:
Updated workflow for Laravel 11

Commit:
Updated workflow for Laravel 11

Commit:
Added separate jobs for Laravel 10 and Laravel 11

Commit:
Added separate jobs for Laravel 9

Commit:
Added package lock to ignore, removed from project

Commit:
Revert changes for package lock

Commit:
Added support for more parameters including docker support

Commit:
Fixed pint issues

Commit:
Fixed pipeline failure detection

Commit:
Fixed pint issues

Commit:
Updated readme

Commit:
Updated readme and tests

Commit:
Updated readme

Commit:
Fixed after rebase

Commit:
Refactoring for strong typing enforce, added phpstan and rector run in pipeline

Commit:
Added rector rules

indy2kro committed:
Merge pull request #1 from indy2kro/v2-upgrade

Refactoring for strong typing enforce, added phpstan and rector

indy2kro commented:
Ack, I have Github's way of adding any new commits to the open PR - right now it became more or less a huge one with a ton of changes (which were not intended to be in a single PR) - although they are .. well, a lot, they are all quite nice, but they do extend the scope quite a bit by also adding strong typing and a few nice enhancements to the pipeline.
All this would be a nice addition to the original project, but it would clearly need an increase in the major version as it breaks compatibility with old custom hooks implementations.

Commit:
Updated phpstan standard, added rector and php insights support

Commit:
Added option for validate_paths

Commit:
Added option for analyzer_chunk_size

indy2kro committed:
Create dependabot.yml

indy2kro committed:
Merge pull request #2 from indy2kro/workflow-improvements

Create dependabot.yml

indy2kro committed:
Create snyk-security.yml

indy2kro committed:
Removed snyk config

indy2kro committed:
Create codacy.yml

indy2kro committed:
Removed codacy config

igorsgm commented:
Thanks for the contributions made so far @indy2kro. I will review and test them all during this week 🙏

Commit:
Fixed param usage

Commit:
Updated chunk size to be configurable per command if needed

Commit:
Fixed rector tests

Commit:
pint fixes

Commit:
pint fixes

Commit:
pint fixes

Commit:
Updated composer reqs for pest plugin laravel

Commit:
Removed Laravel 9 support

Commit:
Updated npm package lock

snyk-bot committed:
fix: upgrade prettier from 2.8.7 to 2.8.8

Snyk has created this PR to upgrade prettier from 2.8.7 to 2.8.8.

See this package in npm:
prettier

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #6 from indy2kro/snyk-upgrade-d8cf1bd2f8f4f3aa71809147d5b2a5eb

[Snyk] Upgrade prettier from 2.8.7 to 2.8.8

indy2kro committed:
Update package-lock.json

indy2kro committed:
Updated npm deps

indy2kro committed:
Updated npm deps

Commit:
Füge Unterstützung für Laravel Sail hinzu

algorhythm committed:
Merge pull request #1 from algorhythm/feature/sail-support

Füge Unterstützung für Laravel Sail hinzu

Commit:
Extend README.md with a "Laravel Sail" section

Commit:
Optimize README.md
Optimize config order

snyk-bot committed:
fix: upgrade eslint-plugin-import from 2.30.0 to 2.31.0

Snyk has created this PR to upgrade eslint-plugin-import from 2.30.0 to 2.31.0.

See this package in npm:
eslint-plugin-import

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #14 from indy2kro/snyk-upgrade-c8e7d7c76fa31db078b134c6acb3b3a6

[Snyk] Upgrade eslint-plugin-import from 2.30.0 to 2.31.0

indy2kro committed:
Merge pull request #13 from algorhythm/master

Add Laravel Sail support

snyk-bot committed:
fix: upgrade eslint-plugin-vue from 9.28.0 to 9.30.0

Snyk has created this PR to upgrade eslint-plugin-vue from 9.28.0 to 9.30.0.

See this package in npm:
eslint-plugin-vue

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #17 from indy2kro/snyk-upgrade-729e6e31cbbca870f9a24eaa3cef8d78

[Snyk] Upgrade eslint-plugin-vue from 9.28.0 to 9.30.0

snyk-bot committed:
fix: upgrade eslint-plugin-vue from 9.30.0 to 9.31.0

Snyk has created this PR to upgrade eslint-plugin-vue from 9.30.0 to 9.31.0.

See this package in npm:
eslint-plugin-vue

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #18 from indy2kro/snyk-upgrade-4ed081fefb819a68fbc826d50b6e0e90

[Snyk] Upgrade eslint-plugin-vue from 9.30.0 to 9.31.0

snyk-bot committed:
fix: upgrade eslint-plugin-vue from 9.31.0 to 9.32.0

Snyk has created this PR to upgrade eslint-plugin-vue from 9.31.0 to 9.32.0.

See this package in npm:
eslint-plugin-vue

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #20 from indy2kro/snyk-upgrade-693c1e206e36f02068850cbbe7e78092

[Snyk] Upgrade eslint-plugin-vue from 9.31.0 to 9.32.0

snyk-bot committed:
fix: upgrade blade-formatter from 1.41.1 to 1.42.0

Snyk has created this PR to upgrade blade-formatter from 1.41.1 to 1.42.0.

See this package in npm:
blade-formatter

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #21 from indy2kro/snyk-upgrade-1ccdbb72389c96dc7461ebafb1a64495

[Snyk] Upgrade blade-formatter from 1.41.1 to 1.42.0

snyk-bot committed:
fix: package.json & package-lock.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-CROSSSPAWN-8303230
- https://snyk.io/vuln/SNYK-JS-INFLIGHT-6095116

indy2kro committed:
updated package.json

indy2kro committed:
Update .eslintrcFixture.js

indy2kro committed:
Update .eslintrcFixture.js

indy2kro committed:
Updated files

indy2kro committed:
Skip test for now

indy2kro committed:
Merge pull request #22 from indy2kro/snyk-fix-7649892480bd83c14c28ebfa1d3c6efc

[Snyk] Security upgrade eslint from 8.57.0 to 9.15.0

indy2kro committed:
Update main.yml

indy2kro committed:
Update main.yml

indy2kro committed:
Updated deps to be more loose with upgrades

indy2kro committed:
Added cron execution for tests

indy2kro committed:
Update main.yml

snyk-bot committed:
fix: package.json & package-lock.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-INFLIGHT-6095116

indy2kro committed:
Merge pull request #23 from indy2kro/snyk-fix-229d80ef893f7c0aef17a13f84ea8ba7

[Snyk] Security upgrade blade-formatter from 1.42.0 to 1.42.1

snyk-bot committed:
fix: upgrade eslint from 9.18.0 to 9.19.0

Snyk has created this PR to upgrade eslint from 9.18.0 to 9.19.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #24 from indy2kro/snyk-upgrade-f4edc88fdb0037397ccc92259b423dc4

[Snyk] Upgrade eslint from 9.18.0 to 9.19.0

Commit:
Updated deps

Commit:
Updated deps

Commit:
Updated deps

Commit:
Updated cache key

Commit:
Added continue-on-error for laravel 12 for now

Commit:
Removed laravel 12 jobs for now

snyk-bot committed:
fix: upgrade blade-formatter from 1.42.1 to 1.42.2

Snyk has created this PR to upgrade blade-formatter from 1.42.1 to 1.42.2.

See this package in npm:
blade-formatter

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade eslint from 9.19.0 to 9.20.0

Snyk has created this PR to upgrade eslint from 9.19.0 to 9.20.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

Commit:
Disable phpinsights security check (unrealiable now)

indy2kro committed:
Merge pull request #26 from indy2kro/snyk-upgrade-6516db885cdf6739c985b165d96381d5

[Snyk] Upgrade eslint from 9.19.0 to 9.20.0

indy2kro committed:
Merge pull request #25 from indy2kro/snyk-upgrade-7b94072229b595c3c6d3d728b687b64d

[Snyk] Upgrade blade-formatter from 1.42.1 to 1.42.2

snyk-bot committed:
fix: upgrade eslint from 9.20.0 to 9.20.1

Snyk has created this PR to upgrade eslint from 9.20.0 to 9.20.1.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #27 from indy2kro/snyk-upgrade-9c44d1f88c1c62967e5921fe1caa52ef

[Snyk] Upgrade eslint from 9.20.0 to 9.20.1

indy2kro committed:
Set fixed eslint version to avoid conflicts

indy2kro committed:
Merge branch 'master' of https://github.com/indy2kro/laravel-git-hooks

indy2kro committed:
Fixed eslint version

indy2kro committed:
Update main.yml

indy2kro committed:
Updated eslint deps

indy2kro committed:
Update ESLintPreCommitHookTest.php

snyk-bot committed:
fix: upgrade eslint from 9.23.0 to 9.24.0

Snyk has created this PR to upgrade eslint from 9.23.0 to 9.24.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #29 from indy2kro/snyk-upgrade-17006ebd4a5dda6a0a6e5658a5082d8e

[Snyk] Upgrade eslint from 9.23.0 to 9.24.0

snyk-bot committed:
fix: upgrade eslint from 9.24.0 to 9.25.1

Snyk has created this PR to upgrade eslint from 9.24.0 to 9.25.1.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #30 from indy2kro/snyk-upgrade-44dfc12a95a6e219c9af9855e252a587

[Snyk] Upgrade eslint from 9.24.0 to 9.25.1

snyk-bot committed:
fix: upgrade eslint from 9.25.1 to 9.26.0

Snyk has created this PR to upgrade eslint from 9.25.1 to 9.26.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #31 from indy2kro/snyk-upgrade-6694cb8bd4e9d7867d3fe2451e8978ae

[Snyk] Upgrade eslint from 9.25.1 to 9.26.0

snyk-bot committed:
fix: upgrade eslint from 9.26.0 to 9.27.0

Snyk has created this PR to upgrade eslint from 9.26.0 to 9.27.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #32 from indy2kro/snyk-upgrade-f89aa81f932862aeb1e63fa858ecdd71

[Snyk] Upgrade eslint from 9.26.0 to 9.27.0

snyk-bot committed:
fix: upgrade eslint-plugin-import from 2.31.0 to 2.32.0

Snyk has created this PR to upgrade eslint-plugin-import from 2.31.0 to 2.32.0.

See this package in npm:
eslint-plugin-import

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #34 from indy2kro/snyk-upgrade-6d47ee5ec0139e26307a8e0002b42cbd

[Snyk] Upgrade eslint-plugin-import from 2.31.0 to 2.32.0

dependabot[bot] committed:
Bump @eslint/plugin-kit in the npm_and_yarn group across 1 directory

---
updated-dependencies:
- dependency-name: "@eslint/plugin-kit"
  dependency-version: 0.3.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

indy2kro committed:
Merge pull request #35 from indy2kro/dependabot/npm_and_yarn/npm_and_yarn-20b018c2ea

Bump @eslint/plugin-kit from 0.3.1 to 0.3.3 in the npm_and_yarn group across 1 directory

snyk-bot committed:
fix: upgrade eslint from 9.27.0 to 9.30.1

Snyk has created this PR to upgrade eslint from 9.27.0 to 9.30.1.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #36 from indy2kro/snyk-upgrade-32dd7e761d1262ef11cf6191e254538a

[Snyk] Upgrade eslint from 9.27.0 to 9.30.1

snyk-bot committed:
fix: upgrade blade-formatter from 0.42.2 to 1.42.3

Snyk has created this PR to upgrade blade-formatter from 1.42.2 to 1.42.3.

See this package in npm:
blade-formatter

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade eslint from 9.30.1 to 9.31.0

Snyk has created this PR to upgrade eslint from 9.30.1 to 9.31.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #39 from indy2kro/snyk-upgrade-6fbe19c6e81fdfeed1df48ea040b0b19

[Snyk] Upgrade eslint from 9.30.1 to 9.31.0

dependabot[bot] committed:
Bump @eslint/plugin-kit in the npm_and_yarn group across 1 directory

Bumps the npm_and_yarn group with 1 update in the / directory: [@eslint/plugin-kit](https://github.com/eslint/rewrite/tree/HEAD/packages/plugin-kit).


Updates `@eslint/plugin-kit` from 0.3.3 to 0.3.4
- [Release notes](https://github.com/eslint/rewrite/releases)
- [Changelog](https://github.com/eslint/rewrite/blob/main/packages/plugin-kit/CHANGELOG.md)
- [Commits](https://github.com/eslint/rewrite/commits/plugin-kit-v0.3.4/packages/plugin-kit)

---
updated-dependencies:
- dependency-name: "@eslint/plugin-kit"
  dependency-version: 0.3.4
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

indy2kro committed:
Merge pull request #37 from indy2kro/dependabot/npm_and_yarn/npm_and_yarn-373e2693b3

Bump @eslint/plugin-kit from 0.3.3 to 0.3.4 in the npm_and_yarn group across 1 directory

indy2kro committed:
Update package-lock.json

indy2kro committed:
Merge pull request #38 from indy2kro/snyk-upgrade-f19078cf342384806080aee8f246ca10

[Snyk] Upgrade blade-formatter from 1.42.2 to 1.42.3

indy2kro committed:
Added mocking for test tools

indy2kro committed:
Update main.yml

indy2kro committed:
Update main.yml

indy2kro committed:
Revert mock changes

indy2kro committed:
Deprecate Laravel 10 support

indy2kro committed:
Update .scrutinizer.yml

indy2kro committed:
Added php-cs-fixer hook

indy2kro committed:
Removed vitest reference

indy2kro committed:
Updated readme

indy2kro committed:
Merge pull request #41 from indy2kro/deprecate-laravel10-support

Deprecate Laravel 10 support

snyk-bot committed:
fix: upgrade eslint from 9.33.0 to 9.34.0

Snyk has created this PR to upgrade eslint from 9.33.0 to 9.34.0.

See this package in npm:
eslint

See this project in Snyk:
https://app.snyk.io/org/indy2kro/project/9f43d42b-212b-4851-b15c-d9c34efe76c1?utm_source=github&utm_medium=referral&page=upgrade-pr

indy2kro committed:
Merge pull request #42 from indy2kro/snyk-upgrade-bedb62a2724f4c0424b754f982dce68b

[Snyk] Upgrade eslint from 9.33.0 to 9.34.0

indy2kro committed:
Reverted upgrade to eslint

indy2kro committed:
Update package-lock.json

indy2kro committed:
Merge branch 'master' of https://github.com/indy2kro/laravel-git-hooks

igorsgm commented:
Thanks for this amazing PR @indy2kro. Incorporated in the latest package version.
